libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteSession.kt

```diff
@@ -74,6 +87,14 @@ class ReplayRouteSession : MapboxNavigationObserver {
             mapboxNavigation?.resetReplayLocation()
             currentRoute = null
             polylineDecodeStream = null
+        } else if (!hasLocationUpdates) {
+            // In order to get route progress updates, we need location updates.
+            // If we don't have any location updates, we don't get route progress updates
```

Comment:
I'm somewhat missing why having location updates equals having route progress. Progress is only received when we're already in active guidance, so shouldn't we process all new routes coming from `RoutesObserver` unless we're already in active guidance (then we can wait for progress)?

Reply:
> I'm somewhat missing why having location updates equals having route progress.

Having location updates means we'll have route progress updates once the route is set. And with route progress update we'll get more info (such as current route geometry index which can be non-zero) than when we just receive the route in `RoutesObserver`, so it makes sense to wait until the first route progress update. 
But no location updates means there will be no route progress, so we shouldn't wait and should start emulating the route from the beginning (which will result in location updates -> route progress updates). But if we don't start emulating, we'll never receive route progress updates.

Reply:
I see your point, but having location updates in the past doesn't mean will get location/progress updates in the future. What if the route was changed after the last point of the previous route, or after the session was stopped/cleared by the user.

Looking at whether we're in active guidance or not, or even better looking if we're currently replaying any points with the `MapboxReplayer` might be a more robust way to handle this case.

Reply:
OK, changed it to rely on played events.

Reply:
The proposed solution suffers from the same exact problem, doesn't it?

I think we're missing something like:
```suggestion
        } else if (!mapboxNavigation.replayer.isPlaying()) {
```

To implement `isPlaying` w need to ensure that:
1. last played event is not equal to last event
2. the job that runs simulation is active

This could added as an `internal` function for now but I see a lot of value in it in general.

Reply:
Yes, my condition would be false, but do we want to accommodate this use case at all? My understanding is that this class' value add is the fact that it can seamlessly switch simulated routes without having to start them from beginning (like refreshes, selecting new alternatives, etc). The only routes which would've been started from beginning are the first route, or a route that comes after completely finishing previous simulation, which sounds reasonable to me.

I'm hesitant to rely on Nav Native producing a progress update without us guaranteeing that a raw location stream exists (which we can do by checking whether player is playing). Otherwise, there are cases (like the one above) which can get the system stuck.

Reply:
OK, makes sense.
The only non-covered case would be starting the simulation not from the beginning of the route. I don't think we have to support it.